rllib/examples/evaluation/custom_evaluation.py

```diff
@@ -0,0 +1,144 @@
+
+"""
+from ray.rllib.algorithms.algorithm import Algorithm
+from ray.rllib.algorithms.algorithm_config import AlgorithmConfig
+from ray.rllib.evaluation.metrics import summarize_episodes
+from ray.rllib.evaluation.worker_set import WorkerSet
+from ray.rllib.examples.env.simple_corridor import SimpleCorridor
+from ray.rllib.utils.test_utils import (
+    add_rllib_example_script_args,
+    run_rllib_example_script_experiment,
+)
+from ray.rllib.utils.typing import ResultDict
+from ray.tune.registry import get_trainable_cls
+
+
+parser = add_rllib_example_script_args(
+    default_iters=50, default_reward=0.7, default_timesteps=50000
+)
+parser.add_argument("--evaluation-parallel-to-training", action="store_true")
+parser.add_argument("--no-custom-eval", action="store_true")
+parser.add_argument("--corridor-length-training", type=int, default=10)
+parser.add_argument("--corridor-length-eval-worker-1", type=int, default=20)
+parser.add_argument("--corridor-length-eval-worker-2", type=int, default=30)
+
+
+def custom_eval_function(algorithm: Algorithm, eval_workers: WorkerSet) -> ResultDict:
+    """Example of a custom evaluation function.
+
+    Args:
+        algorithm: Algorithm class to evaluate.
+        eval_workers: Evaluation WorkerSet.
+
+    Returns:
+        metrics: Evaluation metrics dict.
+    """
+    # Set different env settings for each (eval) EnvRunner. Here we use the EnvRunner's
+    # `worker_index` property to figure out the actual length.
+    # Loop through all workers and all sub-envs (gym.Env) on each worker and call the
+    # `set_corridor_length` method on these.
+    eval_workers.foreach_worker(
+        func=lambda worker: (
+            env.set_corridor_length(
+                args.corridor_length_eval_worker_1
+                if worker.worker_index == 1
+                else args.corridor_length_eval_worker_2
+            )
+            for env in worker.env.envs
+        )
+    )
+
+    # Collect metrics results collected by eval workers in this list for later
+    # processing.
+    rollout_metrics = []
+
+    # For demonstration purposes, run through some arbitrary number of evaluation
+    # round within this one call. Note that this function is called once per
+    # training iteration (`Algorithm.train()` call) OR once per `Algorithm.evaluate()`
+    # (which may be called manually by the user).
+    for i in range(3):
+        print(f"Training iteration {algorithm.iteration} -> evaluation round {i}")
+        # Sample episodes from the EnvRunners AND have them return only the thus
+        # collected metrics.
+        metrics_all_workers = eval_workers.foreach_worker(
+            # Return only the metrics, NOT the sampled episodes (we don't need them
+            # anymore).
+            func=lambda worker: (worker.sample(), worker.get_metrics())[1],
+            local_worker=False,
+        )
+        for metrics_per_worker in metrics_all_workers:
```

Comment:
Can we show here maybe how to sort the metrics from different corridor lengths in the `results` dict? (e.g. such that they show up in different diagrams in wandb/tensorboard)